src/elements/content-sidebar/Sidebar.js

```diff
@@ -1,105 +1,157 @@
-    MARK_NAME_JS_LOADING_ACTIVITY,
-);
-const LoadableSkillsSidebar = SidebarUtils.getAsyncSidebarContent(SIDEBAR_VIEW_SKILLS, MARK_NAME_JS_LOADING_SKILLS);
-const LoadableMetadataSidebar = SidebarUtils.getAsyncSidebarContent(
-    SIDEBAR_VIEW_METADATA,
-    MARK_NAME_JS_LOADING_METADATA,
-);
-
-const Sidebar = ({
-    activitySidebarProps,
-    currentUser,
-    detailsSidebarProps,
-    file,
-    fileId,
-    getPreview,
-    getViewer,
-    metadataSidebarProps,
-    onVersionHistoryClick,
-    selectedView,
-}: Props) => (
-    <React.Fragment>
-        {selectedView === SIDEBAR_VIEW_DETAILS && (
-            <LoadableDetailsSidebar
-                key={fileId}
-                fileId={fileId}
-                onVersionHistoryClick={onVersionHistoryClick}
-                {...detailsSidebarProps}
-                startMarkName={MARK_NAME_JS_LOADING_DETAILS}
-            />
-        )}
-        {selectedView === SIDEBAR_VIEW_SKILLS && (
-            <LoadableSkillsSidebar
-                key={file.id}
-                file={file}
-                getPreview={getPreview}
-                getViewer={getViewer}
-                startMarkName={MARK_NAME_JS_LOADING_SKILLS}
-            />
-        )}
-        {selectedView === SIDEBAR_VIEW_ACTIVITY && (
-            <LoadableActivitySidebar
-                currentUser={currentUser}
-                file={file}
-                onVersionHistoryClick={onVersionHistoryClick}
-                {...activitySidebarProps}
-                startMarkName={MARK_NAME_JS_LOADING_ACTIVITY}
-            />
-        )}
-        {selectedView === SIDEBAR_VIEW_METADATA && (
-            <LoadableMetadataSidebar
-                fileId={fileId}
-                {...metadataSidebarProps}
-                startMarkName={MARK_NAME_JS_LOADING_METADATA}
-            />
-        )}
-    </React.Fragment>
-);
-
-export default Sidebar;
+type State = {
+    isDirty: boolean,
+    isOpen: boolean,
+};
+
+class Sidebar extends React.Component<Props, State> {
+    id: string = uniqueid('bcs_');
+
+    props: Props;
+
+    state: State;
+
+    constructor(props: Props) {
+        super(props);
+
+        const { isLarge } = this.props;
+
+        this.state = {
+            isDirty: false,
+            isOpen: !!isLarge,
+        };
+    }
+
+    componentDidUpdate(prevProps: Props): void {
+        const { fileId, history, isLarge, location }: Props = this.props;
+        const { fileId: prevFileId, isLarge: prevIsLarge }: Props = prevProps;
+        const { isDirty, isOpen }: State = this.state;
+
+        // User navigated to a different file without ever navigating to a tab
+        if (!isDirty && fileId !== prevFileId && location.pathname !== '/') {
+            history.replace({ pathname: '/' });
+        }
+
+        // User resized their viewport without ever navigating to a tab
+        if (!isDirty && isLarge !== prevIsLarge && isLarge !== isOpen) {
+            this.setState({ isOpen: isLarge });
```

Comment:
!!isLarge, wondering why flow didnt suggest that.
What about in the equality? Should we just have it defaultProps?